### PR TITLE
added mse_loss, nll_loss and kl_div_loss

### PR DESCRIPTION
**Pull request by enochkan**
Added more loss functions per https://github.com/ml-explore/mlx/issues/76

### CODE REVIEW

**Review by awni — CHANGES_REQUESTED** [review]
Awesome, thanks for adding these! A few requests then we can merge.

Also could you add the new losses to https://github.com/ml-explore/mlx/blob/main/docs/src/python/nn.rst#loss-functions ?

**Comment by enochkan** [comment]
Can do! Thanks for the feedbacks. I'll be pushing some tests as well.

File: python/mlx/nn/losses.py [hunk 1]
```diff
@@ -25,25 +24,89 @@ def cross_entropy(
...
+        mx.array: The computed NLL loss.
+    """
+    loss = -mx.take_along_axis(logits, targets[..., None], axis).squeeze(-1)
+
+    return _reduce(loss, reduction)
+
+
+def kl_div_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none") -> mx.array:
+    """
+    Computes the Kullback-Leiber divergence loss between logits and targets.
```
Comment (awni) [inline python/mlx/nn/losses.py:88]: Nit: typo `Leibler` with an `l`.

File: python/mlx/nn/losses.py [hunk 2]
```diff
@@ -25,25 +24,89 @@ def cross_entropy(
...
+
+    Args:
+        logits (mx.array): Logits for the distribution p.
+        targets (mx.array): Logits for the distribution q.
+        axis (int, optional): The axis over which to compute softmax. Default: ``-1``.
+        reduction (str, optional): Specifies the reduction to apply to the output:
+          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'none'``.
+
+    Returns:
+        mx.array: The computed Kullback-Leiber divergence loss.
```
Comment (awni) [inline python/mlx/nn/losses.py:98]: Nit: Same typo here: `Leibler`

File: python/mlx/nn/losses.py [hunk 3]
```diff
@@ -25,25 +24,89 @@ def cross_entropy(
...
+    return _reduce(loss, reduction)
+
+
+def kl_div_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none") -> mx.array:
+    """
+    Computes the Kullback-Leiber divergence loss between logits and targets.
+
+    Args:
+        logits (mx.array): Logits for the distribution p.
+        targets (mx.array): Logits for the distribution q.
```
Comment (awni) [inline python/mlx/nn/losses.py:92]: Could you update this to say `log probabilities` for the `targets`

**Review by awni — CHANGES_REQUESTED** [review]
Just a few more quick changes, then we can merge, thanks!!

**Comment by enochkan** [comment]
@awni corrected spellings of Kullback-Leibler divergence and updated docstring for target param.

**Comment by enochkan** [comment]
> Awesome, thanks for adding these! A few requests then we can merge.
> 
> Also could you add the new losses to https://github.com/ml-explore/mlx/blob/main/docs/src/python/nn.rst#loss-functions ?

added.

**Review by awni — APPROVED** [review]
This is awesome, thanks!!
